### PR TITLE
feat: use `foundry-cheatcodes` implementations

### PR DESCRIPTION
**Pull request by DaniPopes**
## Motivation

Unblocks all other cheatcode PRs.

Follow-up to #5998, #6128.

This PR deprecates `foundry-abi`, which will require another follow up to remove completely since it's still used for some trace and log decoding.

## Solution

Replace old Ethers cheatcodes with new Alloy cheatcodes.

## Breaking changes

- `parse*` and `env*`:
  - `bool` values are now parsed as strictly `true|false` instead of `[tT]rue|[fF]alse`
  - `bytes32` values are not padded anymore when encountering less than 32 hex bytes, e.g. `0x0102` won't be padded to `0x0102000000000000000000000000000000000000000000000000000000000000` and instead fail parsing with "invalid string length"
  - Signed integers (not unsigned) don't fail parsing on non-prefixed hex anymore, e.g. `ffff` parses when it should fail because of missing `0x` prefix. This is tracked here: https://github.com/alloy-rs/core/issues/387
- Reverts are now always encoded using the custom error `CheatcodeError(string)`, instead of either the default Solidity revert (`Error(string)`) or an arbitrary signature (`CheatCodeError`)
- Error message strings in general have been changed to be more consistent and less confusing. This breaks tests that check for exact matches on cheatcode error messages, which is not recommended anyway because they may change at any moment.

### CODE REVIEW

File: crates/evm/core/src/backend/error.rs [hunk 1]
```diff
@@ -13,37 +11,41 @@ pub type DatabaseResult<T> = Result<T, DatabaseError>;
...
 pub enum DatabaseError {
-    #[error("Failed to fetch AccountInfo {0:?}")]
+    #[error("{0}")]
+    Message(String),
+    #[error("no cheats available for {0}")]
+    NoCheats(Address),
+    #[error("failed to fetch AccountInfo {0}")]
     MissingAccount(Address),
-    #[error("Could should already be loaded: {0:?}")]
+    #[error("code should already be loaded: {0}")]
```
Comment (mattsse) [inline crates/evm/core/src/backend/error.rs:22]: lol

File: crates/evm/core/src/backend/fuzz.rs [hunk 2]
```diff
@@ -168,7 +168,7 @@ impl<'a> DatabaseExt for FuzzBackendWrapper<'a> {
         transaction: B256,
         env: &mut Env,
         journaled_state: &mut JournaledState,
-        inspector: I,
+        inspector: &mut I,
```
Comment (mattsse) [inline crates/evm/core/src/backend/fuzz.rs:171]: hmm, I don't think this is required because Inspector should be auto_impl for &mut Inspector

**Review by mattsse — COMMENTED** [review]
lfg!!!

one suggestion, otherwise lgtm, pending tests

File: crates/evm/core/src/backend/fuzz.rs (hunk 2 above)
Comment (DaniPopes) [reply crates/evm/core/src/backend/fuzz.rs:171]: used with `cx.state` (`state` is `&mut Cheatcodes`) throws "cannot move out of mutable ref..."
And I prefer this to using with double ref `&mut cx.state`

Comment (mattsse) [reply crates/evm/core/src/backend/fuzz.rs:171]: gotcha

**Review by Evalir — APPROVED** [review]
This is so much better—I think once tests pass we're good to go!

**Comment by refcell** [comment]
🔥

**Pull request merged**